0.preprocess-sites/4.image-and-segmentation-qc.py

```diff
@@ -504,14 +509,47 @@
         )
 
     image_corr_df = (
-        pd.concat(image_corr_list).drop_duplicates(subset="site_location").reset_index()
+        pd.concat(image_corr_list).drop_duplicates(subset="site").reset_index()
+    )
+
+    image_corr_df = image_corr_df.assign(
+        min_corr_value=image_corr_df[corr_qc_cols]
+        .astype(float)
+        .min(skipna=True, axis=1)
     )
 
     for col in corr_qc_cols:
         image_corr_df.loc[(image_corr_df[col] >= correlation_threshold), col] = "pass"
+        if len(image_corr_df[col].unique()) == 1:
+            image_corr_df = image_corr_df.drop(col, axis=1)
 
     if len(image_corr_df.index) > 0:
         corr_output_file = pathlib.Path(results_output, "flagged_correlations.csv")
 
         if check_if_write(corr_output_file, force, throw_warning=True):
             image_corr_df.to_csv(corr_output_file)
+
+# By-well visualization of sites with poor correlations
+if len(corr_qc_cols) > 0:
+    image_corr_df = image_corr_df.merge(loc_df, on="site_location")
+
+    corr_by_well_gg = (
+        gg.ggplot(image_corr_df, gg.aes(x="x_loc", y="y_loc"))
+        + gg.geom_point(gg.aes(fill="min_corr_value"), size=10)
+        + gg.geom_text(gg.aes(label="site"), color="lightgrey")
```

Comment:
checkin that the text should indeed be "site" (include plate name) or "site_location" (just the number)

Reply:
sorry, should say "I am checking in to confirm that..." doh

Reply:
All these `site` names have me really muddled. Looking forward to having v.1 locked down!
All of the plot labels should be just the number, so that's `site_location`

Reply:
me too! Thanks for bearing with these frustrating updates - we'll have a lock soon